frontend/src/app/Settings/Users/ManageUsers.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useState } from "react";
+import { toast } from "react-toastify";
+import { Prompt } from "react-router-dom";
+
+import Alert from "../../commonComponents/Alert";
+import Button from "../../commonComponents/Button";
+import ConditionalWrap from "../../commonComponents/ConditionalWrap";
+import InProgressModal from "./InProgressModal";
+import UserFacilitiesSettingsForm from "./UserFacilitiesSettingsForm";
+import UserRoleSettingsForm from "./UserRoleSettingsForm";
+import { SettingsUser } from "./ManageUsersContainer";
+import { showNotification } from "../../utils";
+
+import "./ManageUsers.scss";
+
+interface Props {
+  loggedInUser: User;
+  users: SettingsUser[];
+  onUpdateUser: (user: SettingsUser) => void;
+}
+
+type SettingsUsers = { [id: string]: SettingsUser };
+
+const ManageUsers: React.FC<Props> = ({
+  users,
+  loggedInUser,
+  onUpdateUser,
+}) => {
+  let settingsUsers: SettingsUsers = users.reduce(
+    (acc: SettingsUsers, user: SettingsUser) => {
+      acc[user.id] = user;
+      return acc;
+    },
+    {}
+  );
+
+  const [usersState, updateUsersState] = useState<SettingsUsers>(settingsUsers);
+  const [activeUserId, updateActiveUserId] = useState<string>(
+    Object.keys(settingsUsers)[0]
+  ); // TODO: unless there is a desired sort algorithm, pick the first user
+  const [nextActiveUserId, updateNextActiveUserId] = useState<string | null>(
+    null
+  );
+  const [showInProgressModal, updateShowInProgressModal] = useState(false);
+
+  function updateUser<T>(
```

Comment:
still learning typescript; not sure if I'm using generics correctly or if this is the best approach